patches/kaggle_secrets.py

```diff
@@ -0,0 +1,59 @@
+import json
+import os
+import urllib.request
+
+_KAGGLE_DEFAULT_URL_BASE = "https://www.kaggle.com"
+_KAGGLE_URL_BASE_ENV_VAR_NAME = "KAGGLE_URL_BASE"
+_KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME = "KAGGLE_USER_SECRETS_TOKEN_KEY"
+
+
+class CredentialError(Exception):
+    pass
+
+
+class BackendError(Exception):
+    pass
+
+
+class UserSecretsClient():
+    GET_USER_SECRET_ENDPOINT = '/requests/GetUserSecretRequest'
+    BIGQUERY_PURPOSE_VALUE = 1
+
+    def __init__(self):
+        url_base_override = os.getenv(_KAGGLE_URL_BASE_ENV_VAR_NAME)
+        self.url_base = (url_base_override or _KAGGLE_DEFAULT_URL_BASE)
+        # Follow the OAuth 2.0 Authorization standard (https://tools.ietf.org/html/rfc6750)
+        jwt_token = os.getenv(_KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME)
+        if jwt_token is None:
+            raise CredentialError(
+                f'A JWT Token is required to use the UserSecretsClient, but none found in environment variable {_KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME}')
+        self.headers = {'Content-type': 'application/json',
+                        'Authorization': 'Bearer {}'.format(jwt_token)}
+
+    def _make_get_request(self, request_body):
+        request_params = urllib.parse.urlencode(request_body)
+        url = f'{self.url_base}{self.GET_USER_SECRET_ENDPOINT}?{request_params}'
+        req = urllib.request.Request(url, headers=self.headers)
+        with urllib.request.urlopen(req) as response:
+            response_json = json.loads(response.read())
+            return response_json
+
+    def getUserSecret(self, secret_label: str):
+        request_body = {
+            'SecretLabel': secret_label
+        }
+        response_json = self._make_get_request(request_body)
+        if 'Secret' not in response_json:
+            raise BackendError(
+                'Unexpected response from UserSecrets service.')
```

Comment:
I'd also remove the word UserSecret from error messages.

Reply:
I've removed it, but just curious why? The user will be using the UserSecrets library, so this shouldn't be very surprising.

Reply:
It still shows the old message for me for some reason on github?
I'd remove because this particular name is implementation specific, and also why give more information than necessary here.

Reply:
Ah I only removed it from the other method, not this one, will fix it now. 

What do you mean implementation specific? This class is a client to the UserSecrets service?

Reply:
I mean that the name we use on server side is impl specific, and don't have to be known (unlike the endpoint we expose, etc.). In any case this is also a nit.